lib/validation.rb

```diff
@@ -51,4 +51,10 @@ def self.validate_os_user_name(os_user_name)
     msg = "OS user name must only contain lowercase letters, numbers, hyphens and underscore and cannot start with a number or hyphen. It also have max length of 32."
     fail ValidationFailed.new({user: msg}) unless os_user_name.match(ALLOWED_OS_USER_NAME_PATTERN)
   end
+
+  def self.validate_storage_volumes(storage_volumes)
+    if storage_volumes.empty?
+      fail "At least one storage volume is required."
+    end
```

Comment:
nit: Validation module raises `ValidationFailed` in general
```suggestion
    fail ValidationFailed.new({storage_volumes: "At least one storage volume is required."}) if storage_volumes.empty?
```